packages/frontend/src/hooks/useBalance.tsx

```diff
@@ -1,64 +1,28 @@
-import { useState, useRef, useEffect, useCallback } from 'react'
-import { BigNumber } from 'ethers'
+import { useQuery } from 'react-query'
 import { Token } from '@hop-protocol/sdk'
 import Network from 'src/models/Network'
-import logger from 'src/logger'
-import { useInterval } from 'react-use'
 import { Addressish } from 'src/models/Address'
 
-const useBalance = (
-  token: Token | undefined,
-  network: Network | undefined,
-  address: Addressish
-) => {
-  const [balance, setBalance] = useState<BigNumber>()
-  const [loading, setLoading] = useState(false)
-  const currentToken = useRef<Token>()
-  const currentNetwork = useRef<Network>()
-
-  const getBalance = useCallback(() => {
-    let isSubscribed = true
-    const _getBalance = async () => {
-      if (token && network && address) {
-        if (
-          (currentNetwork.current && !network.eq(currentNetwork.current)) ||
-          (currentToken.current && !token.eq(currentToken.current))
-        ) {
-          setLoading(true)
-        }
-
-        const _balance = await token.balanceOf(address.toString())
+async function fetchBalance(token: Token, address: Addressish) {
+  return token.balanceOf(address?.toString())
+}
 
-        if (isSubscribed) {
-          setBalance(_balance as BigNumber)
-          setLoading(false)
-        }
-      } else {
-        if (isSubscribed) {
-          setBalance(undefined)
-          setLoading(false)
-        }
-      }
-      if (isSubscribed) {
-        currentToken.current = token
-        currentNetwork.current = network
+const useBalance = (token?: Token, network?: Network, address?: Addressish) => {
+  const { isLoading, isError, data, error } = useQuery(
+    ['balance', token?.symbol, network?.slug, address?.toString],
+    () => {
+      if (token && address) {
+        return fetchBalance(token, address)
       }
+    },
+    {
+      enabled: !!token?.symbol && !!address?.toString,
+      refetchInterval: 8e3,
```

Comment:
I think we can update this to 5 seconds (`5 * 1000`) to be consistent with other pollers